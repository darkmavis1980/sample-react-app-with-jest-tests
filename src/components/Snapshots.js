import React, {Component} from 'react';
import SnapshotItem from './SnapshotItem';

class Snapshots extends Component {
  constructor(props) {
    super(props);

    this.state = {
      list: [
        'steve',
        'frank',
        'alex'
      ]
    }
  }

  render() {
    return (
      <div>
        <h3>Snaps</h3>
        {this.state.list.map((item, index) => {
          return (<SnapshotItem key={index} name={item} />)
        })}
      </div>
    );
  }
}

export default Snapshots;