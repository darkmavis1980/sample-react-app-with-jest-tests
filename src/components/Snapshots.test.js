import React from 'react';
import Snapshots from './Snapshots';
import renderer from 'react-test-renderer';

describe('(Component) Snapshots', () => {

  const tree = renderer.create(
    <Snapshots />
  ).toJSON();

  describe('render()', () => {
    it('should render the component', () => {
      expect(tree).toMatchSnapshot();
    });
  });

});