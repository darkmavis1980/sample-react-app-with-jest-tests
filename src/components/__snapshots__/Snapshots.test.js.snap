// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Snapshots render() should render the component 1`] = `
<div>
  <h3>
    Snaps
  </h3>
  <div>
    steve
  </div>
  <div>
    frank
  </div>
  <div>
    alex
  </div>
</div>
`;
